elastic/kibana
# [Discover/Reporting] Fix potential time drift with relative time when requesting a report

## Summary

Partially resolves https://github.com/elastic/kibana/issues/114193

Since we allow relative times in search source (https://github.com/elastic/kibana/pull/110459) there is a possibility that when a user requests a report from Discover, this report will be generated an unknown amount of time after the user requested it in the UI.

This PR adds logic to allow Discover to provide either a relative or absolute time filtered SearchSource.

## How to test

1. Start Kibana & ES
2. Create some sample data (via the UI possibly) and create a discover search with a narrow time filter and a small number of items rendered in the UI
3. Wait a minute or so until you are outside the time window shown in the UI, then request a report.
4. The report should contain the correct values (i.e., the values reflected in the UI).

## Release note

TODO

### Checklist
